src/Processors/Streaming/ReplayStepTransform.cpp

```diff
@@ -0,0 +1,92 @@
+#include <elf.h>
+#include <Processors/Streaming/ReplayStepTransform.h>
+#include <Common/ProtonCommon.h>
+#include <Processors/ISimpleTransform.h>
+#include <Processors/ProcessorID.h>
+#include <base/ClockUtils.h>
+#include <base/sleep.h>
+
+namespace DB
+{
+namespace Streaming
+{
+ReplayStepTransform::ReplayStepTransform(const Block & header, float replay_speed_)
+    : ISimpleTransform(header, header, true, ProcessorID::ReplayStepTransformID)
+    , replay_speed(replay_speed_)
+{
+    tp_time_index = header.getPositionByName(ProtonConsts::RESERVED_APPEND_TIME);
+}
+
+void ReplayStepTransform::transform(Chunk & input_chunk, Chunk & output_chunk)
+{
+    if (time_up)
+    {
+        if (input_chunk.rows())
+        {
+            /// get the time of this chunk.
+            const auto & this_batch_time = input_chunk.getColumns()[tp_time_index]->get64(0);
+            interval = static_cast<Int64>((last_batch_time == 0 ? 0 : this_batch_time - last_batch_time) / replay_speed);
+            last_batch_time = this_batch_time;
+            time_up = false;
+        }
+        else    /// in case of the empty chunk.
+        {
+            output_chunk.swap(input_chunk);
+            time_up = true;
+            has_input = false;
+        }
+    }
+    else
+    {
+        sleepForMilliseconds(interval > 500 ? 500 : interval);
```

Comment:
use `std::this_thread::sleep_for`

Reply:
use a constexpr var `max_wait_interval = 500` instead of hard const value anywhere.

Reply:
why use the `std::this_thread::sleep_for`?

Reply:
> why use the `std::this_thread::sleep_for`?

we prefer using the standard library.